dsc_lib/src/discovery/command_discovery.rs

```diff
@@ -320,26 +318,27 @@ impl ResourceDiscovery for CommandDiscovery {
                         }
                     }
                 }
-                self.adapters = adapters;
-                self.resources = resources;
+                extend_adapters(adapters);
+                extend_resources(resources);
             },
             DiscoveryKind::Extension => {
-                self.extensions = extensions;
+                extend_extensions(extensions);
             }
         }
 
         Ok(())
     }
 
     fn discover_adapted_resources(&mut self, name_filter: &str, adapter_filter: &str) -> Result<(), DscError> {
-        if self.resources.is_empty() && self.adapters.is_empty() {
+        if resources_is_empty() && adapters_is_empty() {
             self.discover(&DiscoveryKind::Resource, "*")?;
         }
 
-        if self.adapters.is_empty() {
+        if adapters_is_empty() {
             return Ok(());
         }
 
+        let adapters = get_adapters();
```

Comment:
Is there any possible race condition between this (where it's cloned) and where it's used a few times below?

Reply:
All the helper functions that return are clones so the race condition (which doesn't exist today since it's single threaded essentially) is that the function is working off a clone and the actual source might have changed.  The mutex, currently, isn't required, but then the Rust compiler requires `unsafe {}` code to access globals due to the risk of a race condition.